File: Binary Search/nthRoot.java
class Solution {
    public int nthRoot(int n, int m) {
        // code here
        int low = 0;
        int high = m/n + 1;
        
        while(low<=high){
            int mid = (low+high)/2;
            int val = (int) Math.pow(mid, n);
            if(val == m){
                return mid;
            } else if(val>m){
                high = mid-1;
            } else{
                low = mid+1;
            }
        }
        return -1;
    }
}

File: Binary Search/nthRoot.py
def NthRoot(n: int, m: int) -> int:
    # Write Your Code Here
    left = 0
    right = m//n + 1

    while(left<=right):
        mid = (left+right)//2
        ans = 1
        for i in range(1,n+1):
            ans*=mid
        if ans == m:
            return mid
        elif ans > m:
            right = mid-1
        else:
            left = mid+1
    
    return -1

    pass